### PR TITLE
fix: stripe connect atom for teams

### PR DESCRIPTION
**Pull request by Ryukemeister**
## What does this PR do?

- Previously we were only handling stripe integration for an individual user. This PR adds teams integration for stripe so that stripe is enabled for teams and individual event types.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
This can be tested in the examples app we have in `packages/platform/examples/base`

### CODE REVIEW

File: apps/api/v2/src/modules/users/users.repository.ts [hunk 1]
```diff
@@ -248,4 +249,46 @@ export class UsersRepository {
...
+              select: {
+                id: true,
+                isOrganization: true,
+              },
+            },
+          },
+        },
+      },
+    });
+  }
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/users/users.repository.ts:293]: we should already have the different repository functions necessary to check if a user is OrgOwner/OrgOwner/TeamOwner/TeamAdmin, in v2/src/modules/auth/guards/roles/roles.guard.ts

File: apps/api/v2/src/modules/stripe/controllers/stripe.controller.ts [hunk 2]
```diff
@@ -99,6 +102,16 @@ export class StripeController {
   @HttpCode(HttpStatus.OK)
   @ApiOperation({ summary: "Check stripe connection" })
   async check(@GetUser() user: UserWithProfile): Promise<StripCredentialsCheckOutputResponseDto> {
-    return await this.stripeService.checkIfStripeAccountConnected(user.id);
+    return await this.stripeService.checkIfIndividualStripeAccountConnected(user.id);
+  }
+
+  @Get("/check/:teamId")
```
Comment (Ryukemeister) [inline apps/api/v2/src/modules/stripe/controllers/stripe.controller.ts:108]: Individual and team checks for stripe have been separated into two distinct endpoints now, having both checks under one endpoint was causing a mess

File: apps/api/v2/src/modules/stripe/stripe.service.ts [hunk 3]
```diff
@@ -102,18 +126,39 @@ export class StripeService {
...
+  async checkIfUserHasAdminAccessToTeam(teamId: number, userId: number) {
+    const teamMembership = await this.membershipRepository.findMembershipByTeamId(teamId, userId);
+    const hasAdminAccessToTeam = teamMembership?.role === "ADMIN" || teamMembership?.role === "OWNER";
+
+    if (!hasAdminAccessToTeam) {
+      throw new BadRequestException("You must be team owner or admin to do this");
+    }
+  }
+
+  async validateStripeCredentials(
```
Comment (Ryukemeister) [inline apps/api/v2/src/modules/stripe/stripe.service.ts:150]: This is a dumb function that checks validity of stripe credentials, this way we can pass in credentials for team as well as individuals depending on the check we want to do.

File: packages/app-store/_components/AppCard.tsx [hunk 4]
```diff
@@ -81,7 +81,7 @@ export default function AppCard({
             </p>
           </div>
           <div className="ml-auto flex items-center space-x-2">
-            {app.credentialOwner && (
+            {app.credentialOwner && !isPlatform && (
```
Comment (Ryukemeister) [inline packages/app-store/_components/AppCard.tsx:84]: Avatar can't be displayed for a platform user, hence needed to add this `isPlatform` check

**Review by ThyMinimalDev — APPROVED** [review]

**Pull request merged**